pixl_dcmd/src/pixl_dcmd/_database.py

```diff
@@ -34,13 +35,13 @@
 
 def add_hashed_identifier_and_save_to_db(
     existing_image: Image, hashed_value: str
-) -> Image:
+) -> Optional[Image]:
     PixlSession = sessionmaker(engine)
     with PixlSession() as pixl_session, pixl_session.begin():
         existing_image.hashed_identifier = hashed_value
         pixl_session.add(existing_image)
 
-        updated_image: Image = (
+        updated_image: Optional[Image] = (
             pixl_session.query(Image)
             .filter(
                 Image.accession_number == existing_image.accession_number,
```

Comment:
Been a while since I thought about this, but why are we querying for one or none here? Can there be a time where we don't find an image by the identifiers?

Reply:
I don't know what this code is trying to achieve. It seems to be querying the DB for the row that it just added/updated. The return value is not used anyway in the one place this method is called. So unless the query has a side-effect it could probably be deleted altogether.
Maybe we need a separate task for tidying up our usage of sqlalchemy, since the session creation is also a bit copy n pastey?

Reply:
Now issue #385